plantseg/io/h5.py

```diff
@@ -109,12 +109,28 @@ def create_h5(path: str,
         f[key].attrs['element_size_um'] = voxel_size
 
 
-def list_keys(path: str) -> list[str]:
+def list_keys(path):
     """
-    returns all datasets in a h5 file
+    List all keys in a h5 file
+    Args:
+        path: path to the h5 file
+
+    Returns:
+        list of keys
     """
+    def recursive_find_keys(f, base='/'):
+        _list_keys = []
+        for key, dataset in f.items():
+            if isinstance(dataset, h5py.Group):
+                new_base = f"{base}{key}/"
```

Comment:
this won't work if you start with anything other than root, e.g. `recursive_find_keys(f, base='/labels')`. Using `os.path.join(base, key)` here and below (when appending to `_list_keys`) should always be safe

Reply:
This is correct for `zarr,` and I just fixed it :) 

But h5 using `os.path.join` will not work on Windows. H5 implements a posix internal path "x/y" that does not changes on different platforms. So `os.path.join` would only work in posit OS but not on Windows. 
Also, `recursive_find_keys` is in the inner scope of  `list_keys,` so it will always start with the root.